doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,1685 @@
+    tuning_not_accepted_set_time               = STR           ; tuning not accepted flag set time
+    invalid_channel_num_set_time               = STR           ; invalid channel number flag set time
+    tuning_complete_set_time                   = STR           ; tuning complete flag set time
+```
+
+#### 2.3.5 Transceiver status data to store module and data path flag clear time
+
+The `TRANSCEIVER_STATUS_FLAG_CLEAR_TIME` table stores the clear time for the transceiver flag.
+
+lane_num: Represents lane number of the field. The lane number is an integer value that ranges from 1 to 8.
+
+```plaintext
+    ; Defines Transceiver Status info for a port
+    key                                        = TRANSCEIVER_STATUS_FLAG_CLEAR_TIME|ifname        ; Flag information for module on port
+    ; field                                    = value
+    datapath_firmware_fault_clear_time           = STR           ; datapath (DSP) firmware fault clear time
+    module_firmware_fault_clear_time             = STR           ; module firmware fault clear time
+    module_state_changed_clear_time              = STR           ; module state changed clear time
+    txfault{lane_num}_clear_time                 = STR           ; tx fault flag on media lane {lane_num} clear time
+    txlos_hostlane{lane_num}_clear_time          = STR           ; tx loss of signal flag on host lane {lane_num} clear time
+    txcdrlol_hostlane{lane_num}_clear_time       = STR           ; tx clock and data recovery loss of lock flag on host lane {lane_num} clear time
+    tx_eq_fault{lane_num}_clear_time             = STR           ; tx equalization fault flag on host lane {lane_num} clear time
+    rxlos{lane_num}_clear_time                   = STR           ; rx loss of signal flag on media lane {lane_num} clear time
+    rxcdrlol{lane_num}_clear_time                = STR           ; rx clock and data recovery loss of lock flag on media lane {lane_num} clear time
+    target_output_power_oor_clear_time           = STR           ; target output power out of range flag clear time
+    fine_tuning_oor_clear_time                   = STR           ; fine tuning  out of range flag clear time
+    tuning_not_accepted_clear_time               = STR           ; tuning not accepted flag clear time
+    invalid_channel_num_clear_time               = STR           ; invalid channel number flag clear time
+    tuning_complete_clear_time                   = STR           ; tuning complete flag clear time
+```
+
+### 2.4 Transceiver PM data
+
+The `TRANSCEIVER_PM` table stores the performance monitoring data of the transceiver. This table is exists only for C-CMIS transceivers.
+
+```plaintext
+    ; Defines Transceiver PM information for a port
+    key                          = TRANSCEIVER_PM|ifname            ; information of PM on port
+    ; field                      = value 
+    prefec_ber_avg               = FLOAT                            ; prefec ber avg
```

Comment:
we could expand this to media and host prefec BER and uncorr frames, even if this is C-CMIS only.

Media BER and UC frames info comes from page 34h, host BER and UC frames come from page 3Ah